testing/smoke/managed/main.tf

```diff
@@ -0,0 +1,297 @@
+terraform {
+  required_version = ">= 1.1.8, < 2.0.0"
+  required_providers {
+    ec = {
+      source  = "elastic/ec"
+      version = "0.5.1"
+    }
+  }
+}
+
+provider "ec" {}
+
+module "ec_deployment" {
+  source = "../../infra/terraform/modules/ec_deployment"
+  region = var.region
+
+  deployment_template    = "gcp-compute-optimized-v2"
+  deployment_name_prefix = "smoke-upgrade"
+
+  apm_server_size = "1g"
+
+  elasticsearch_size       = "4g"
+  elasticsearch_zone_count = 1
+
+  stack_version = var.stack_version
+  tags          = merge(local.ci_tags, module.tags.tags)
+}
+
+locals {
+  image_owners = {
+    "ubuntu-bionic-18.04-amd64-server" = "099720109477" # canonical
+    "ubuntu-focal-20.04-amd64-server"  = "099720109477" # canonical
+    "ubuntu-jammy-22.04-amd64-server"  = "099720109477" # canonical
+    "debian-10-amd64"                  = "136693071363" # debian
+    "debian-11-amd64"                  = "136693071363" # debian
+    "amzn2-ami-kernel-5.10"            = "137112412989" # amazon
+    "RHEL-7"                           = "309956199498" # Red Hat
+    "RHEL-8"                           = "309956199498" # Red Hat
+    "RHEL-9"                           = "309956199498" # Red Hat
+  }
+  image_ssh_users = {
+    "ubuntu-bionic-18.04-amd64-server" = "ubuntu"
+    "ubuntu-focal-20.04-amd64-server"  = "ubuntu"
+    "ubuntu-jammy-22.04-amd64-server"  = "ubuntu"
+    "debian-10-amd64"                  = "admin"
+    "debian-11-amd64"                  = "admin"
+    "amzn2-ami-kernel-5.10"            = "ec2-user"
+    "RHEL-7"                           = "ec2-user"
+    "RHEL-8"                           = "ec2-user"
+    "RHEL-9"                           = "ec2-user"
+  }
```

Comment:
Instead of duplicating this bit, should we create a new module where all these are defined? Then we can use the outputs on both `managed` and `supported-os`

Reply:
Done!